nemo/collections/nlp/models/language_modeling/megatron_base_model.py

```diff
@@ -377,13 +384,21 @@ def _validate_config(self):
             with open_dict(self.cfg):
                 self.cfg.sequence_parallel = False
 
-        if (
-            self.cfg.get('gradient_accumulation_fusion', False)
-            and self.cfg.get('pipeline_model_parallel_size', 1) == 1
-        ):
-            logging.info("Gradient accumulation fusion can only be used with pipeline parallel size > 1.")
-            with open_dict(self.cfg):
-                self.cfg.gradient_accumulation_fusion = False
+        # Gradient accumulation fusion does not work with our baseline implementaiton of
+        # async grad allreduce. This should be fixed!
+        # For now we must disable it whenever using the baseline implementaion.
+        # The distributed adam from apex does work with gradient accumulation fusion.
+        distributed_fused_adam = self.cfg.optim.get('name', 'fused_adam') == 'distributed_fused_adam'
+        pipeline_model_parallel_size = self.cfg.get('pipeline_model_parallel_size', 1)
+        data_parallel_size = app_state.data_parallel_size
+
+        if self.cfg.get('gradient_accumulation_fusion', False):
+            if data_parallel_size > 1 and pipeline_model_parallel_size == 1 and not distributed_fused_adam:
+                logging.info(
+                    "When not using pipeline model parallel, gradient accumulation fusion can only be used with distributed_fused_adam."
+                )
+                with open_dict(self.cfg):
+                    self.cfg.gradient_accumulation_fusion = False
```

Comment:
Can we do this in a separate function? or rename this function so that it indicates we are both validating args as well as overriding some of them if needed?